### PR TITLE
Explain analyze

### PR DESCRIPTION
**Pull request by Simpleshaikh1**
I made sure that I have my database set up with the schema and data from my previous projects.

I run the following query to add an extra column to the owners table:
-- Add an email column to owners table
ALTER TABLE owners ADD COLUMN email VARCHAR(120);

I run the following statements to add data to your database (executing them might take a few minutes):

INSERT INTO visits (animal_id, vet_id, date_of_visit) SELECT * FROM (SELECT id FROM animals) animal_ids, (SELECT id FROM vets) vets_ids, generate_series('1980-01-01'::timestamp, '2021-01-01', '4 hours') visit_timestamp;

-- This added 2.500.000 owners with full_name = 'Owner <X>' and email = 'owner_<X>@email.com' (~2min approx.)
insert into owner (full_name, email) select 'Owner ' || generate_series(1,2500000), 'owner_' || generate_series(1,2500000) || '@mail.com';

The following queries are taking too much time (1 sec = 1000ms can be considered as too much time for database query). I tried them on my machine to confirm it:

SELECT COUNT(*) FROM visits where animal_id = 4;
SELECT * FROM visits where vet_id = 2;
SELECT * FROM owners where email = 'owner_18327@mail.com';

I used EXPLAIN ANALYZE on the previous queries to check what is happening. Took screenshots of them 

I decrease the execution time of the first query.

I improved execution time of the other two queries.

I checked if they help by running EXPLAIN ANALYZE.  took a screenshot of the EXPLAIN ANALYZE result showing that time actually decreased.

All the changes I made to my database schema were added to the schema.sql file and commited in a new pull request.

bellow are the screenshots with EXPLAIN ANALYZE results (before and after I improved the speed) for each of the 3 problematic queries.

261380119-499a4aa6-2b62-4a00-8156-a7fb15853686
261482442-79b17940-ea62-48ae-972c-f1dea0ad30bb
261382026-de4e02f8-b35b-48b6-953b-a7fcf8e04562

another thing I noticed is that the execution time changes from every execution, Thought is still bellow 1ms

Also, I want to make it known that I am working solo this week, I don't have a coding partner. Below is the screenshot of my dashboard with no coding partner attached to me.

Screenshot (54)

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -65,3 +64,25 @@ CREATE TABLE visits (
...
+
+CREATE INDEX idx_animal_id ON visits(animal_id);
+EXPLAIN ANALYZE SELECT COUNT(*) FROM visits WHERE animal_id = 4;
+
+CREATE INDEX idx_vet_id ON visits(vet_id);
+EXPLAIN ANALYZE SELECT COUNT(*) FROM visits WHERE vet_id = 4;
+
+
+CREATE INDEX idx_email ON owners(email);
+EXPLAIN ANALYZE SELECT COUNT(*) FROM visits WHERE email = 4;
```
Comment (V-Blaze) [inline schema.sql]: - You have done great so far :+1: :+1:  but there are a few requirements and changes you need to make before moving to the next milestone. :wave: 

- [ ] Currently, looking at the screenshot in your pull request description, it is a bit difficult to see the results of the queries before the database improvements and screenshots after the improvements. I'll suggest you take the screenshots individually and place them accordingly to make things easier, Looking at the screenshot below, you can see how its arranged to show before and after improvement for each query. :+1: :+1: 

- - Screenshot reference:
Selection_005

File: schema.sql [hunk 2]
```diff
@@ -65,3 +64,25 @@ CREATE TABLE visits (
...
+
+SELECT COUNT(*) FROM visits where animal_id = 4;
+EXPLAIN ANALYZE SELECT COUNT(*) FROM visits WHERE animal_id = 4;
+
+
+SELECT * FROM visits where vet_id = 2;
+SELECT * FROM owners where email = 'owner_18327@mail.com';
+
+CREATE INDEX idx_animal_id ON visits(animal_id);
+EXPLAIN ANALYZE SELECT COUNT(*) FROM visits WHERE animal_id = 4;
```
Comment (V-Blaze) [inline schema.sql]: - [ ] looking at the screenshot for the `EXPLAIN ANALYZE SELECT COUNT(*) FROM visits WHERE animal_id = 4;` before adding an index, the `Planning and Execution time` is below `1000ms`. According to the requirement, you need to keep populating the database with the data provided until the `Execussion time is above `1000ms`, then you work on it to reduce it. I'll suggest you take this into consideration/. :+1: :100:

File: schema.sql (hunk 1 above)
Comment (V-Blaze) [inline schema.sql]: - [ ] You did great creating the indexes inside the `schema.sql` file. but the `EXPLAIN ANALYZE` are supposed to be inside the `querries.sql` file. I'll suggest you move all the three queeires `EXPLAIN ANALYZE SELECT COUNT(*)  FROM visits WHERE animal_id = 4;`, `EXPLAIN ANALYZE SELECT COUNT(*) FROM visits WHERE vet_id = 2;`, `SELECT * FROM owners where email = 'owner_18327@mail.com';` there. :smile: :+1: 

- [ ] Also your last 2 queries are different from the one that was in the requirement. this is what you have currently:
- - 
Selection_006
 And this is what is on the requirement:
- -
Selection_007

**Review by V-Blaze — CHANGES_REQUESTED** [review]
## Hi @Simpleshaikh1 

## Required changes 🖊️ 
# Almost there 🟡 🟦 🟨 

almost-there


Great work so far, You have come a long way, keep moving forward👏 👍  
You've done well implementing the project requirements, but there are still some issues I think you need to improve on. 😄 👋 


 ## Highlights
- Used `INDEX` to improve execution time✔
- Correct Git flow👋 
-  Descriptive pull request summary  ✔

 ## Required Changes ♻️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by thecodechaser — APPROVED** [review]
## STATUS: APPROVED 🟢

## Hi @Simpleshaikh1👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 🎉
- Good job with the performance audit. ✔️
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**